app/window.py

```diff
@@ -873,21 +873,10 @@ def refresh(self):
   def setTextBuffer(self, textBuffer):
     app.log.info('setTextBuffer')
     #self.normalize()
-    if self.textBuffer is not None:
-      app.history.set(['files', self.textBuffer.fullPath, 'cursor'],
-          (self.textBuffer.penRow, self.textBuffer.penCol))
-    # TODO(dschuyler): Do we need to restore positions and selections here?
     self.controller.setTextBuffer(textBuffer)
     Window.setTextBuffer(self, textBuffer)
+    self.textBuffer.restoreUserHistory()
```

Comment:
I think that this idea is really good for when you snap back to the cursor when your scrollView is at a different location. This currently is only implemented for when you find an expression and the result is not in the view. What do you think about making this optimalCursorRow/Col to apply also when your cursor is out of view and you type/use the arrow keys? 

I still think when specifically resuming a file, the position should be saved at the same spot? Or maybe we can cap it at the optimalCursorPos defaults. For example, if the cursor is at below the preference (.28) and I save and resume editing, the view will now be scrolled to .28, but if the cursor was at .48 (percentage of rows from the bottom), then it would stay at .48